gcp/api/server.py

```diff
@@ -254,7 +278,8 @@ def _is_version_affected(affected_packages,
               affected_package.package.ecosystem) != ecosystem):
         continue
 
-    if purl and purl != affected_package.package.purl:
+    if purl and not (affected_package.package.purl and _match_purl(
```

Comment:
nit: can we just push the logic of handling an empty purl into `_match_purl`? This makes this logic a bit easier to comprehend.

Reply:
Ah I just realised I skimmed this too quickly and the types here don't match. Can skip this comment.